### PR TITLE
test_pg_autoscale_flag.py: Added test to verify autoscale flag 

### PR DESCRIPTION
**Pull request by pdhiran**
Signed-off-by: Pawan Dhiran <pdhiran@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pdhiran** [comment]
Logs : http://magna002.ceph.redhat.com/ceph-qe-logs/pawan/cephci-run-LCER7W/

**Comment by pdhiran** [comment]
Suite file logs : http://magna002.ceph.redhat.com/ceph-qe-logs/pawan/cephci-run-RLO6T4/

File: tests/rados/test_pg_autoscale_flag.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    # Turning the autoscale flag back on. All the setting made earlier should be reverted
+    cmd = "ceph osd pool unset noautoscale"
+    pool_status = rados_obj.run_ceph_command(cmd=cmd)
+
+    # sleeping for 5 seconds as the command takes some time to affect the status of pools
+    time.sleep(5)
+
+    for entry in pool_status:
+        if entry["pg_autoscale_mode"] == "off":
```
Comment (SrinivasaBharath) [inline tests/rados/test_pg_autoscale_flag.py:103]: "ceph osd pool unset noautoscale" makes the flag is set to be off.

File: tests/rados/test_pg_autoscale_flag.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
+"""
+test Module to :
+1. Create pool
+2. write objects
+3. Turn on/off the autoscale flag and verifying the pool status
+4. Verify the configuration changes when the flag gets set/unset
+7. Delete pools
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply tests/rados/test_pg_autoscale_flag.py]: Corrected

File: tests/rados/test_pg_autoscale_flag.py (hunk 1 above)
Comment (pdhiran) [reply tests/rados/test_pg_autoscale_flag.py:103]: ` ceph osd pool unset noautoscale ` makes the flag is set to be off( no autoscale for pg's restriction removed ).  That means pg_autoscaler module will be active again. Hence iterating through all the pools to check if ` pg_autoscale_mode ` is off. It should be enabled once the flag is unset.

**Review by tintumathew10 — COMMENTED** [review]
Looks good

**Comment by pdhiran** [comment]
@neha-gangadhar , @psathyan , Can you please review?

**Review by neha-gangadhar — APPROVED** [review]
Looks good to me

**Review by tintumathew10 — APPROVED** [review]

**Pull request merged**